constellations/gemini-admin/src/classes/classEdit.js

```diff
@@ -9,23 +9,44 @@ import { OkayModal } from "../modals/okayModal.js";
 import { YesNoModal } from "../modals/yesnoModal.js";
 
 export class ClassEditPage extends React.Component {
-    state = {
-        isEdit: false,
-
-        // class object
-        oldClassId: "",
-        inputClassKey: "",
-        inputTimeString: "",
-
-        selectProgramId: "",
-        selectSemesterId: "",
-        selectLocationId: "",
-
-        programs: [],
-        semesters: [],
-        locations: [],
-        sessions: [],
-    };
```

Comment:
keep this format and add your additional variables. This "automatically" defines the constructor function and initializes the state variable. It's the more modern away to define state in ReactJs